Entities/HDFirearm.cpp

```diff
@@ -844,6 +846,9 @@ void HDFirearm::Update()
                     pParticle->SetVel(m_Vel + particleVel);
                     pParticle->SetRotAngle(particleVel.GetAbsRadAngle() + (m_HFlipped ? -c_PI : 0));
 					pParticle->SetHFlipped(m_HFlipped);
+					if (lifeVariation != 0 && pParticle->GetLifetime() != 0) {
+						pParticle->SetLifetime(std::max(static_cast<int>(pParticle->GetLifetime() * (1.0F + (particleCountMax > 1 ? lifeVariation - (lifeVariation * 2.0F * (pRound->ParticleCount() / particleCountMax)) : lifeVariation * RandomNormalNum()))), 1));
```

Comment:
This part seems odd to me: `lifeVariation - (lifeVariation * 2.0F * (pRound->ParticleCount() / particleCountMax))`
1. particleCountMax is pRound->ParticleCount(), so you're just doing lifeVariation * 2 * 1. Or am I mistaken?
2. What is the intent of this complex stuff exactly? Why not just have life variation apply normally if there's more than one particle in the round?

Reply:
ParticleCount is set up pretty shittily where it decrements for each particle, so `ParticleCount()` actually returns the amount after the decrement which happens in `PopNextParticle()` or whatever, all during the same frame

so the temp value `particleCountMax` stores the actual full particle count in the round

Reply:
Ahhh, that makes sense. I'd missed that about particle count.
So the idea here is that each subsequent particle has slightly more lifetime than the previous one, which I guess ends up with them all dying at the same distance?

Reply:
Consistently at the same rate, yeah. (not all consequently shot particles dying at the same distance, which you somehow made it sound like lol)

Dunno if it's the most intuitive way of treating it, it would probably make more sense for it to go only negative so at least you could treat the static lifetime value as the maximum... but whatever